ballista/scheduler/src/test_utils.rs

```diff
@@ -783,6 +783,47 @@ pub fn assert_failed_event(job_id: &str, collector: &TestMetricsCollector) {
     assert!(found, "{}", "Expected failed event for job {job_id}");
 }
 
+pub fn revive_graph_and_complete_next_stage(graph: &mut ExecutionGraph) -> Result<usize> {
```

Comment:
this helper method just finds the next stage that has an available task and completes the stage. This was required because we create scans or memtables with 2 partitions, which result in 2 tasks in the 1st stage. [This](https://github.com/apache/datafusion/issues/12611) only causes regression when the input doesn't have multiple partitions